aten/src/THNN/generic/SpatialUpSamplingBicubic.c

```diff
@@ -2,225 +2,33 @@
-  THTensor_(resize4d)(output,
-          THTensor_(size)(input, 0),
-          THTensor_(size)(input, 1),
-          output_height, output_width);
-  THTensor_(zero)(output);
-  scalar_t *idata = input->data<scalar_t>();
-  scalar_t *odata = output->data<scalar_t>();
-
-  // Special case: input/output same size, just copy
-  if (input_height == output_height && input_width == output_width) {
-    for (int output_y = 0; output_y < output_height; output_y++) {
-      for (int output_x = 0; output_x < output_width; output_x++) {
-        const scalar_t* in = &idata[output_y * input_width + output_x];
-        scalar_t* out = &odata[output_y * output_width + output_x];
-        for (int c = 0; c < channels; ++c) {
-          out[0] = in[0];
-          in += input_width * input_height;
-          out += output_width * output_height;
-        }
-      }
-    }
-    c10::raw::intrusive_ptr::decref(input);
-    return;
-  }
-
-  // Bicubic interpolation
-  const accreal height_scale = linear_upsampling_compute_scale<accreal>(
-    input_height,
-    output_height,
-    align_corners);
-  const accreal width_scale = linear_upsampling_compute_scale<accreal>(
-    input_width,
-    output_width,
-    align_corners);
-
-  for (int output_y = 0; output_y < output_height; output_y++) {
-    for (int output_x = 0; output_x < output_width; output_x++) {
-      scalar_t* in = idata;
-      scalar_t* out = odata;
-
-      const scalar_t real_x = width_scale * output_x;
-      int input_x = real_x;
-      const scalar_t t_x = real_x - input_x;
-
-      const scalar_t real_y = height_scale * output_y;
-      int input_y = real_y;
-      const scalar_t t_y = real_y - input_y;
-
-      for (int c = 0; c < channels * nbatch; c++) {
-        scalar_t coefficients[4];
-
-        // Interpolate 4 times in the x direction
-        for (int i = 0; i < 4; i++) {
-          coefficients[i] = cubic_interp1d<scalar_t>(
-            upsampling_get_value_bounded<scalar_t>(
-              in, input_width, input_height, input_x - 1, input_y - 1 + i),
-            upsampling_get_value_bounded<scalar_t>(
-              in, input_width, input_height, input_x + 0, input_y - 1 + i),
-            upsampling_get_value_bounded<scalar_t>(
-              in, input_width, input_height, input_x + 1, input_y - 1 + i),
-            upsampling_get_value_bounded<scalar_t>(
-              in, input_width, input_height, input_x + 2, input_y - 1 + i),
-            t_x
-          );
-        }
-
-        // Interpolate in the y direction using x interpolations
-        out[output_y * output_width + output_x] = cubic_interp1d<scalar_t>(
-          coefficients[0],
-          coefficients[1],
-          coefficients[2],
-          coefficients[3],
-          t_y
-        );
-
-        // Move to next channel
-        in += input_width * input_height;
-        out += output_width * output_height;
-      }
-    }
-  }
-
-  c10::raw::intrusive_ptr::decref(input);
+  return;
```

Comment:
I don't like this. You should throw an error. You can use `AT_ERROR` to do that.

Reply:
so it means if anyone tries to use the old functions for cpu it should raise an error, is that correctly?

is this message ok? `This function is deprecated, please use it from ATen`

Reply:
Yes, exactly!